### ISSUE TITLE
Add implicit

### ISSUE THREAD

**Issue by trebconnell** [Microsoft/GSL#120]
[GSL.Util](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#-gslutil-utilities) now contains a macro *implicit* that you can place in from of implicit conversions to call them out.

We should define this, and use it in the gsl classes

**Comment by trebconnell** [comment]
Using this in all gsl classes is dependent on #101 as it needs to live in a util header other than gsl.h

**Comment by neilmacintosh** [comment]
Sure. Feel free to offer a PR for the change.

**Comment by gdr-at-ms** [comment]
Did we reach a decision that we want `implicit` as macro?  That would be surprising... I seem to remember someone  suggested attribute `[[implicit]]` -- which is arguably more verbose, but petfectly fine with respect to the guidelines themselves and scoping rules.



-------- Original message --------

**Comment by neilmacintosh** [comment]
My memory was also that we liked the attribute better, but the Core Guidelines says macro at the moment. @trebconnell you might want to hold off on any PR until Gaby and I resolve with the editors of the Core Guidelines what the preferred option really is.

**Comment by trebconnell** [comment]
Okay, let me know once it's resolved.  Macro's definitely a sketchy
solution, but it's nice to have similar syntax to explicit.  If I were to
do the macro I'd at least have it expand to an attribute so that variables
named implicit might produce semi-legible errors.

**Comment by neilmacintosh** [comment]
Opened 
isocpp/CppCoreGuidelines/#282 to track over at Core Guidelines.

**Comment by neilmacintosh** [comment]
Attribute it is! See https://github.com/isocpp/CppCoreGuidelines/issues/282.

Implicit should be said `[[implicit]]`.
